### ISSUE TITLE
Generated password when create new account not respect spec about length limit for password

### ISSUE THREAD

**Issue by PhilippeAccorsi** [tracim/tracim#1617]
## Bug description & symptom

I have created a new account on tracim. User receive an email but the password is not accepted by the backend, error 2001:
~~~
{"message": "Validation error of input data", "details": {"password": ["Length must be between 6 and 512."]}, "code": 2001}
~~~

## Prerequisites

- Notification by email is activated
- smtp account is configured

## How to reproduce

Its difficult because this is automatic generation of password. 

## Impacted components

- frontend
- backend

## Extra information

Here is the exemple I have received:
~~~
Hello ****,

Someone invited you to join TRACIM.

Your credentials are:

    login: *******@*******.****
    password: K`{x 

Let start to discuss, share files, agenda and documentation with collaborators by logging into your shared space: http://localhost:6543/ui/login

note: as soon as you are connected, we strongly recommend that you change your password and delete this email.

Enjoy your day :)

Suricat', your digital assistant

--
Suricat', the bot
TRACIM
http://localhost:6543
powered by tracim software - https://www.tracim.fr
~~~

And this is the html code

~~~
<p>Hello ******,</p>

<p>Someone invited you to join <i>TRACIM</i>.</p>

<p>Your credentials are:</p>

<ul>
    <li>login: ******@****.****</li>
    <li>password: K`{x<O9mq$rm</li>
</ul>

<p>
    Let start to discuss, share files, agenda and documentation with collaborators by logging into your shared space: <a href="http://localhost:6543/ui/login">http://localhost:6543/ui/login</a>
</p>
<p>note: as soon as you are connected, we strongly recommend that you change your password and delete this email.</p>
<p>Enjoy your day :)</p>
<p>Suricat', your digital assistant</p>

<pre>
--
Suricat', the bot
TRACIM
<a href="http://localhost:6543">http://localhost:6543</a>
powered by tracim software - <a href="https://www.tracim.fr">https://www.tracim.fr</a>
</pre>
~~~

## Diagnostic

Specials chars like < can broke html rendering of this email.

## Expected

- [ ] encode the password when rendered in HTML.

Edit:
 - change Diagnostic and expected.

**Comment by buxx** [comment]
@PhilippeAccorsi Problem come from password characters: `K`{x<O9mq$rm`. We can see a `<` char, so html rendering of your email is broken.

Password generation is done with this code:

``` python
def password_generator(
        length: int=DEFAULT_PASSWORD_GEN_CHAR_LENGTH,
        chars: str=ALLOWED_AUTOGEN_PASSWORD_CHAR
) -> str:
    """
    :param length: length of the new password
    :param chars: characters allowed
    :return: password as string
    """
    return ''.join(random.choice(chars) for char_number in range(length))
```

Where `ALLOWED_AUTOGEN_PASSWORD_CHAR` is:

``` python
import string

ALLOWED_AUTOGEN_PASSWORD_CHAR = string.ascii_letters + \
                                string.digits + \
                                string.punctuation
```

And `string.punctuation` value is:

``` python
punctuation = """!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~"""
```

So specials chars like `<` can broke html rendering of this email. Two way to fix:

* Encode password in html when write it in html. Result here will be `K`{x&lt;O9mq$rm`
* Do not use these (`!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~`) special chars in generated password

@PhilippeAccorsi (don't forgot to update `Diagnostic` part of email)

**Comment by lebouquetin** [comment]
The right solution is to encode the password when rendered in HTML.

**Comment by PhilippeAccorsi** [comment]
damien_accorsi 11:47
List of availble characters: `!#$%&()*+,-./:;=?@[]_{}`

**Comment by inkhey** [comment]
@lebouquetin Just did some check, we need also to encode properly other fields in mail template. That's actually a bit messy as part of html is generated by code and other part is generated by template.

**Comment by buxx** [comment]
@PhilippeAccorsi changes done in 4c567bc8 as you requested.

**Comment by PhilippeAccorsi** [comment]
Verified.

**Comment by PhilippeAccorsi** [comment]
We have discovered a small bug with `&` in password. This is fixed by #1678 .
